### PR TITLE
Vulnerabilities drits

### PR DESCRIPTION
**Pull request by avrahams**
Add /v1_users_notifications_ vulnerabilities API 
Support pagination for unique values 
Support arrays expansion for unique value
Add delete by query - DELETE /query with V2ListRequest
Enable V2LIstRequest query for Collaboration-Configs

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -55,6 +55,68 @@ type DocContent interface {
...
+	v.UpdatedTime = updatedTime.UTC().Format(time.RFC3339)
+}
+
+func (v *VulnerabilityNotification) GetUpdatedTime() *time.Time {
+	if v.UpdatedTime == "" {
+		return nil
+	}
+	updatedTime, err := time.Parse(time.RFC3339, v.UpdatedTime)
+	if err != nil {
+		return nil
```
Comment (RinaO1234) [inline types/types.go:97]: why hiding the error? if no need to return it let's at least log something

File: types/api_info_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		name          string
+		arrayPaths    []string
+		path          string
+		wantIsArray   bool
+		wantArrayPath string
+		wantSubPath   string
+	}{
+		{
+			name:          "path matches array path",
+			arrayPaths:    []string{"users", "orders"},
```
Comment (RinaO1234) [inline types/api_info_test.go:18]: let's check empty path case and empty arrayPaths case

File: testers_test.go [hunk 3]
```diff
@@ -294,19 +294,39 @@ func testDeleteByName[T types.DocContent](suite *MainTestSuite, basePath, namePa
 
 func testDeleteBulkDeleteByGUID[T types.DocContent](suite *MainTestSuite, basePath string, testDocs []T, compareOpts ...cmp.Option) {
 	//test bulk delete by Ids in body
```
Comment (RinaO1234) [inline testers_test.go:296]: rewrite to avoid core duplication (consider table driven test or defining the patterns as method

Comment (RinaO1234) [reply testers_test.go:296]: repeating delete in "testDeleteBulkDeleteByGUID" is by mistake?

File: testers_test.go [hunk 4]
```diff
@@ -294,19 +294,39 @@ func testDeleteByName[T types.DocContent](suite *MainTestSuite, basePath, namePa
...
 	//test bulk delete by Ids in query
-	newDocs = testBulkPostDocs(suite, basePath, testDocs, commonCmpFilter)
+	newDocs = testBulkPostDocs(suite, basePath, testDocs, compareOpts...)
 	docsIds = []string{}
 	for i := range newDocs {
 		docsIds = append(docsIds, newDocs[i].GetGUID())
 	}
 	testBulkDeleteByGUIDWithBody(suite, basePath, docsIds)
+
+	//test delete by query by Ids in query
```
Comment (RinaO1234) [inline testers_test.go:311]: same comment is repeated, can we specify the case we are trying to test or divide each case to test func?

File: testers_test.go [hunk 5]
```diff
@@ -294,19 +294,39 @@ func testDeleteByName[T types.DocContent](suite *MainTestSuite, basePath, namePa
...
 	for i := range newDocs {
 		docsIds = append(docsIds, newDocs[i].GetGUID())
 	}
 	testBulkDeleteByGUIDWithBody(suite, basePath, docsIds)
 	//test bulk delete by Ids in query
-	newDocs = testBulkPostDocs(suite, basePath, testDocs, commonCmpFilter)
+	newDocs = testBulkPostDocs(suite, basePath, testDocs, compareOpts...)
 	docsIds = []string{}
 	for i := range newDocs {
 		docsIds = append(docsIds, newDocs[i].GetGUID())
```
Comment (RinaO1234) [inline testers_test.go:307]: redundant append because array length is known

File: types/api_info.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+func GetAPIInfo(apiName string) *APIInfo {
+	if apiInfo, ok := apisInfo[apiName]; ok {
+		return apiInfo
+	}
+	return nil
+}
+
+func SetAPIInfo(apiName string, apiInfo *APIInfo) {
+	apisInfo[apiName] = apiInfo
```
Comment (RinaO1234) [inline types/api_info.go:26]: can we check that map is allocated?

File: types/api_info.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+func (s *SchemaInfo) IsArrayPath(path string) (isArray bool, arrayPath, subPath string) {
```
Comment (RinaO1234) [inline types/api_info.go:29]: consider renaming to GetArrayPathDetails shince the function doesn't return only boolean

File: types/api_info.go [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+	DBCollection string     `json:"dbCollection"`
+	Type         string     `json:"type"`
+	Schema       SchemaInfo `json:"schema"`
+}
+
+type SchemaInfo struct {
+	ArrayPaths []string `json:"arrayPaths,omitempty"`
+}
+
+func GetAPIInfo(apiName string) *APIInfo {
```
Comment (RinaO1234) [inline types/api_info.go:18]: why do we return the reference if this is only get?

File: types/api_info.go [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+func SetAPIInfo(apiName string, apiInfo *APIInfo) {
+	apisInfo[apiName] = apiInfo
+}
+
+func (s *SchemaInfo) IsArrayPath(path string) (isArray bool, arrayPath, subPath string) {
+	for _, ap := range s.ArrayPaths {
+		if strings.HasPrefix(path, ap) {
+			isArray = true
+			arrayPath = ap
+			if strings.HasPrefix(path, arrayPath+".") {
```
Comment (RinaO1234) [inline types/api_info.go:34]: and you can do avoid this additional check: 
                        arrayPath := strings.TrimPrefix(path, arrayPath+".")
			if arrayPath != path { // This checks if the path was actually prefixed with arrayPath+"."
				subPath = arrayPath		}
			return

File: service_test.go [hunk 10]
```diff
@@ -1347,587 +1346,3 @@ func (suite *MainTestSuite) TestAttackChainsConfigs() {
 
 	testDeleteDocByGUID(suite, consts.AttackChainsPath, &expectedAttackChainConfigAfterUpdate, commonCmpFilter, ignoreTime)
 }
-
-func (suite *MainTestSuite) TestUsersNotificationsCache() {
```
Comment (RinaO1234) [inline service_test.go:1351]: why are we deleting

File: types/types.go (hunk 1 above)
Comment (avrahams) [reply types/types.go:97]: This method GET is used only for test 
ad all these function are used only inside the service 
the only one that sets the time is the DB package

File: testers_test.go (hunk 3 above)
Comment (avrahams) [reply testers_test.go:296]: This was a mistake - changed to testBulkDeleteByGUIDWithQuery
This test just post documnets and delete them by guid in body, then by guid in query thn by query, then by empty query 
not need for tables here - it does it to all relevant apis

File: types/api_info.go (hunk 6 above)
Comment (avrahams) [reply types/api_info.go:26]: it is allocated in L5

File: types/api_info.go (hunk 9 above)
Comment (avrahams) [reply types/api_info.go:34]: This of a case where you get "myarray" and one where you get "myarray.element.name"
the first check if the array is in path 
the secode one check for the sub path in the array

File: types/api_info.go (hunk 8 above)
Comment (avrahams) [reply types/api_info.go:18]: So I can return nil if it does not exist - but I don't need this function for now - se deleted

**Review by RinaO1234 — APPROVED** [review]

File: types/api_info.go (hunk 6 above)
Comment (RinaO1234) [reply types/api_info.go:19]: any case it will be called without allocation?

**Pull request merged**